jetty-core/jetty-compression/jetty-compression-api/src/main/java/org/eclipse/jetty/compression/Compression.java

```diff
@@ -0,0 +1,355 @@
+            return false;
+        }
+
+        // check the accept encoding header
+        if (!headers.contains(HttpHeader.ACCEPT_ENCODING, getEncodingName()))
+        {
+            if (LOG.isDebugEnabled())
+                LOG.debug("{} excluded not {} acceptable {}", this, getEncodingName(), headers);
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Acquire a {@link RetainableByteBuffer} that is managed by this {@link Compression} implementation
+     * which is suitable for compressed output from an {@link EncoderSink} or compressed input from a {@link DecoderSource}.
+     *
+     * <p>
+     * It is recommended to use this method so that any compression specific details can be
+     * managed by this Compression implementation (such as ByteOrder or buffer pooling)
+     * </p>
+     *
+     * <p>
+     * The size of the buffer comes from {@link Compression} implementation.
+     * </p>
+     *
+     * @return the ByteBuffer suitable for this compression implementation.
+     */
+    public abstract RetainableByteBuffer acquireByteBuffer();
+
+    /**
+     * Acquire a {@link RetainableByteBuffer} that is managed by this {@link Compression} implementation
+     * which is suitable for compressed output from an {@link EncoderSink} or compressed input from a {@link DecoderSource}.
+     *
+     * <p>
+     * It is recommended to use this method so that any compression specific details can be
+     * managed by this Compression implementation (such as ByteOrder or buffer pooling)
+     * </p>
+     *
+     * @param length the requested size of the buffer
+     * @return the ByteBuffer suitable for this compression implementation.
+     */
+    public abstract RetainableByteBuffer acquireByteBuffer(int length);
+
+    /**
+     * Get an etag with suffix that represents this compression implementation.
+     *
+     * @param etag an etag
+     * @return the etag with compression suffix
+     */
+    public String etag(String etag)
+    {
+        if (StringUtil.isEmpty(EtagUtils.ETAG_SEPARATOR))
+            return etag;
+        int end = etag.length() - 1;
+        if (etag.charAt(end) == '"')
+            return etag.substring(0, end) + etagSuffixQuote;
+        return etag + etagSuffix;
+    }
+
+    public int getBufferSize()
+    {
+        return bufferSize;
+    }
+
+    public void setBufferSize(int size)
+    {
+        if (size <= 0)
+            throw new IllegalArgumentException("Invalid buffer size: " + size);
+        this.bufferSize = size;
+    }
+
+    public ByteBufferPool getByteBufferPool()
+    {
+        return this.byteBufferPool;
+    }
+
+    public void setByteBufferPool(ByteBufferPool byteBufferPool)
+    {
+        this.byteBufferPool = byteBufferPool;
+    }
+
+    /**
+     * Get the container being used for common components.
+     *
+     * @return the container for common components
+     */
+    public Container getContainer()
```

Comment:
The container can be Server, in case of CompressionHandler, or various WebSocket components in case of websocket, or Jetty HttpClient in case of client usage.   I know of no way to discover these variations easily in doStart().